lib/React.js

```diff
@@ -39,3 +39,12 @@ var useReducer = function(reducer, initialState) {
   var tuple = React.useReducer(jsReducer, initialState);
   return caml_js_to_array(tuple);
 };
+
+// Provides: useEffect
+// Requires: React, caml_js_wrap_callback, caml_js_from_array
+var useEffect = function(effect) {
+  var callback = caml_js_wrap_callback(
+    effect === 0 ? null : caml_js_from_array(effect)[0]
```

Comment:
This is kind of tricky. What this code is doing is assuming that the effect callback itself is optional, but what we want is to convert the optional params that exist "inside" the callback.

In order to do so, one can create an inline function here, that is the one that is passed to `React.useEffect`. And then inside that callback, convert the required parameters.

I am just working on a similar example for `forwardRef`:

https://github.com/jchavarri/rroo/blob/6f6ac6b39bcd2d884a81b8e4f39de3d2869c6742/lib/React.js#L61-L67

Hope this helps!

Reply:
Feel free to copy the `optFromJs` from that branch, or inline it as it was before, we can consolidate later on 👍

Reply:
Yay! I got the js implementation working correctly. Thank you for the pointer! Btw on the code snippet you shared above, is the `jsCallback` function supposed to take `ref` as a param instead of `jsRef`? It appears variable `ref` is unused.

Reply:
Great!

> It appears variable `ref` is unused.

Hm... you're right, good catch! I will open an issue to revisit `forwardRef`.

As an aside, I am thinking if we could leverage TypeScript or eslint to keep track of these very simple errors. I'll create a new issue for that too 👍